chia/rpc/util.py

```diff
@@ -105,11 +115,9 @@ def tx_endpoint(
     merge_spends: bool = True,
 ) -> Callable[[RpcEndpoint], RpcEndpoint]:
     def _inner(func: RpcEndpoint) -> RpcEndpoint:
-        async def rpc_endpoint(self, request: dict[str, Any], *args, **kwargs) -> dict[str, Any]:
-            if TYPE_CHECKING:
-                from chia.rpc.wallet_rpc_api import WalletRpcApi
-
-                assert isinstance(self, WalletRpcApi)
+        async def rpc_endpoint(
+            self: WalletRpcApi, request: dict[str, Any], *args: object, **kwargs: object
+        ) -> dict[str, Any]:
```

Comment:
do we care to use `EndpointResult` here?

Reply:
Sure, done.